components/timezone/src/ixdtf.rs

```diff
@@ -326,11 +328,14 @@ impl<'a> Intermediate<'a> {
         };
         let time_zone_id = mapper.iana_bytes_to_bcp47(iana_identifier);
         let date = Date::<Iso>::try_new_iso(self.date.year, self.date.month, self.date.day)?;
+        let Some(nanosecond) = self.time.fraction.to_nanoseconds() else {
+            return Err(ParseError::InvalidFraction);
+        };
         let time = Time::try_new(
             self.time.hour,
             self.time.minute,
             self.time.second,
-            self.time.nanosecond,
+            nanosecond,
         )?;
```

Comment:
Ideally there'd only be one code path that parses the IXDTF struct into `Time`, and it should be `Time::try_from_ixdtf_record`. Please try to use that (also for `Date` ideally).

Reply:
I updated `Time` with a `try_from_time_record` method to address this.

I'm not entirely clear what you had meant by also "for `Date` ideally." Unless the idea was to expose two methods: `Time::try_from_time_record` and `Date::try_from_date_record` on the referenced types. So if you were thinking of that or a bit of a different approach, let me know.

Reply:
yeah I was thinking of that, but it's more complicated for `Date` because it's in a different crate, so what you have now is fine

Reply:
Yeah that's why I had paused on it. I wasn't the biggest fan of exposing a public `Date::try_from_date_record` on `Date`.